cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/ResizedRootBlockDeviceMappingProvider.java

```diff
@@ -0,0 +1,152 @@
+                if (describeLaunchTemplateVersionsResponse.hasLaunchTemplateVersions()) {
+                    Optional<LaunchTemplateVersion> defaultLaunchTemplate = describeLaunchTemplateVersionsResponse.launchTemplateVersions().stream()
+                            .filter(LaunchTemplateVersion::defaultVersion)
+                            .findFirst();
+                    if (defaultLaunchTemplate.isPresent()) {
+                        return createBlockDeviceMappingBasedOnDefaultLaunchTemplate(ec2Client, updatableFields, cloudStack, defaultLaunchTemplate.get());
+                    } else {
+                        LOGGER.warn("'describeLaunchTemplateVersions' doesn't have any default template: {}", describeLaunchTemplateVersionsResponse);
+                        return List.of();
+                    }
+                } else {
+                    LOGGER.warn("'describeLaunchTemplateVersions' returned with empty result, skipping resizing root disk. Response: {}",
+                            describeLaunchTemplateVersionsResponse);
+                    return List.of();
+                }
+            } catch (Exception e) {
+                LOGGER.error("Couldn't fetch LaunchTemplateVersions, skipping updating root disk size", e);
+                return List.of();
+            }
+        } else {
+            LOGGER.info("No {} found in updatableFields, skipping resizing root disk", LaunchTemplateField.ROOT_DISK);
+            return List.of();
+        }
+    }
+
+    private List<LaunchTemplateBlockDeviceMappingRequest> createBlockDeviceMappingBasedOnDefaultLaunchTemplate(AmazonEc2Client ec2Client,
+            Map<LaunchTemplateField, String> updatableFields, CloudStack cloudStack, LaunchTemplateVersion defaultLaunchTemplate) {
+        String rootDeviceName = volumeBuilderUtil.getRootDeviceName(cloudStack, ec2Client);
+        List<LaunchTemplateBlockDeviceMapping> blockDeviceMappings = defaultLaunchTemplate.launchTemplateData().hasBlockDeviceMappings()
+                ? new ArrayList<>(defaultLaunchTemplate.launchTemplateData().blockDeviceMappings())
+                : new ArrayList<>();
+        Optional<LaunchTemplateBlockDeviceMapping> rootBlockDeviceMapping = blockDeviceMappings.stream()
+                .filter(blockDeviceMapping -> blockDeviceMapping.deviceName().equals(rootDeviceName))
+                .findFirst();
+        LOGGER.info("rootBlockDeviceMapping {} for rootDeviceName [{}} in LaunchTemplateVersion {}.",
+                rootBlockDeviceMapping, rootDeviceName, defaultLaunchTemplate);
+        if (rootBlockDeviceMapping.isPresent()
+                && !Integer.valueOf(updatableFields.get(LaunchTemplateField.ROOT_DISK)).equals(rootBlockDeviceMapping.get().ebs().volumeSize())) {
+            return constructBlockDeviceMappingRequestWithResizedRootDisk(updatableFields, blockDeviceMappings, rootBlockDeviceMapping.get());
```

Comment:
I would go with greater-than instead of not-equal.

Reply:
that logic is not here. At this point we expect that the incoming value is the right one. eg if later this would be available on public API you might decide to downscale root disk size if you created your cluster with larger disks than required.